code/modules/spells/spell_types/hivemind.dm

```diff
@@ -0,0 +1,465 @@
+			if(!can_target(H))
+				continue
+			if(!hive.hivemembers.Find(H))
+				possible_targets += H
+	else
+		possible_targets = hive.hivemembers.Copy()
+		if(range)
+			possible_targets &= view_or_range(range, user, selection_type)
+
+	var/mob/living/carbon/human/H = input("Choose the target for the spell.", "Targeting") as null|mob in possible_targets
+	if(!H)
+		revert_cast()
+		return
+	targets += H
+	perform(targets,user=user)
+
+/obj/effect/proc_holder/spell/target_hive/hive_add
+	name = "Assimilate Vessel"
+	desc = "We silently add an unsuspecting target to the hive."
+	selection_type = "view"
+	action_icon_state = "add"
+
+	charge_max = 300
+	range = 4
+	target_external = 1
+
+/obj/effect/proc_holder/spell/target_hive/hive_add/cast(list/targets, mob/living/user = usr)
+	var/mob/living/carbon/human/target = targets[1]
+	var/datum/antagonist/hivemind/hive = user.mind.has_antag_datum(/datum/antagonist/hivemind)
+	var/success = FALSE
+
+	if(target.mind && target.stat != DEAD)
+		if(!target.has_trait(TRAIT_MINDSHIELD))
+			to_chat(user, "<span class='notice'>We begin linking our mind with [target.name]!</span>")
+			if(do_mob(user,user,70))
+				to_chat(user, "<span class='notice'>Our mind is ready to connect [target.name] to the Hive!</span>")
+				if((target in view(range)) && do_mob(user,target,30))
+					to_chat(user, "<span class='notice'>[target.name] was added to the Hive!</span>")
+					success = TRUE
+					hive.add_to_hive(target)
+				else
+					to_chat(user, "<span class='notice'>The connection between [target.name] has been disrupted!</span>")
+			else
+				to_chat(user, "<span class='notice'>We fail to connect to [target.name].</span>")
+		else
+			to_chat(user, "<span class='warning'>Powerful technology protects [target.name]'s mind.</span>")
+	else
+		to_chat(user, "<span class='notice'>We detect no neural activity in this body.</span>")
+	if(!success)
+		revert_cast()
+
+/obj/effect/proc_holder/spell/target_hive/hive_remove
+	name = "Release Vessel"
+	desc = "We silently remove a nearby target from the hive. We must be close to their body to do so."
+	selection_type = "view"
+	action_icon_state = "remove"
+
+	charge_max = 100
+	range = 4
+
+/obj/effect/proc_holder/spell/target_hive/hive_remove/cast(list/targets, mob/living/user = usr)
+	var/mob/living/carbon/human/target = targets[1]
+
+	var/datum/antagonist/hivemind/hive = user.mind.has_antag_datum(/datum/antagonist/hivemind)
+	hive.hivemembers -= target
+	hive.calc_size()
+	to_chat(user, "<span class='notice'>We remove [target.name] from the hive</span>")
+
+/obj/effect/proc_holder/spell/target_hive/hive_see
+	name = "Hive Vision"
+	desc = "We use the eyes of one of our vessels. Use again to look through our own eyes once more."
+	action_icon_state = "see"
+
+	charge_max = 150
+
+/obj/effect/proc_holder/spell/target_hive/on_lose(mob/user)
+	if(active)
+		user.reset_perspective(null)
+
+/obj/effect/proc_holder/spell/target_hive/hive_see/cast(list/targets, mob/living/user = usr)
```

Comment:
Probably need to add a sanity check on the spell process to make sure the target isn't qdel'd and to reset vision if it happens. The user can currently do that manually but it's definitely neater for it to work automatically